### PR TITLE
Forward dogstatsd packets to another UDP server

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Adds optional forwarding of dogstatsd packets to another UDP server

### Motivation

Agent5 functionality used by some customers.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -137,6 +137,8 @@ func init() {
 	Datadog.SetDefault("dogstatsd_stats_buffer", 10)
 	Datadog.SetDefault("dogstatsd_expiry_seconds", 300)
 	Datadog.SetDefault("dogstatsd_origin_detection", false) // Only supported for socket traffic
+	Datadog.SetDefault("statsd_forward_host", "")
+	Datadog.SetDefault("statsd_forward_port", "")
```
Comment (masci) [inline pkg/config/config.go]: can we use an `int`, same as `dogstatsd_port`?

File: pkg/dogstatsd/server.go [hunk 2]
```diff
@@ -80,6 +81,25 @@ func NewServer(metricOut chan<- *metrics.MetricSample, eventOut chan<- metrics.E
...
+
+		forwardAddress := fmt.Sprintf("%s:%s", forwardHost, forwardPort)
+
+		con, err := net.Dial("udp", forwardAddress)
+
+		if err != nil {
+			log.Warnf("could not connect to statsd forward host : %s", err)
+		} else {
+			forwardedPacketChanel := make(chan *listeners.Packet, 100)
+			s.packetIn = forwardedPacketChanel
```
Comment (masci) [inline pkg/dogstatsd/server.go]: can we assign `s.packetIn = make(chan *listeners.Packet, 100)` directly?

**Review by masci — CHANGES_REQUESTED** [review]
Good stuff! Left a couple of comments but this LGTM.

**Review by masci — APPROVED** [review]

**Comment by olivielpeau** [comment]
@arbll could you document the new options in the `config_template` ? unless we don't want to advertise the feature

**Pull request merged**